editor/src/document/properties_panel_message_handler.rs

```diff
@@ -429,25 +454,77 @@ fn node_section_transform(layer: &Layer) -> LayoutRow {
 }
 
 fn node_section_fill(fill: &Fill) -> LayoutRow {
-	LayoutRow::Section {
-		name: "Fill".into(),
-		layout: vec![LayoutRow::Row {
-			name: "".into(),
-			widgets: vec![
-				WidgetHolder::new(Widget::TextLabel(TextLabel {
-					value: "Color".into(),
-					..TextLabel::default()
-				})),
-				WidgetHolder::new(Widget::Separator(Separator {
-					separator_type: SeparatorType::Related,
-					direction: SeparatorDirection::Horizontal,
-				})),
-				WidgetHolder::new(Widget::TextInput(TextInput {
-					value: fill.color().rgba_hex(),
-					on_update: WidgetCallback::new(|text_input| PropertiesPanelMessage::ModifyFill { value: text_input.value.clone() }.into()),
-				})),
+	match fill {
+		Fill::Solid(color) => LayoutRow::Section {
+			name: "Fill".into(),
+			layout: vec![LayoutRow::Row {
+				name: "".into(),
+				widgets: vec![
+					WidgetHolder::new(Widget::TextLabel(TextLabel {
+						value: "Color".into(),
+						..TextLabel::default()
+					})),
+					WidgetHolder::new(Widget::Separator(Separator {
+						separator_type: SeparatorType::Related,
+						direction: SeparatorDirection::Horizontal,
+					})),
+					WidgetHolder::new(Widget::TextInput(TextInput {
+						value: color.rgba_hex(),
+						on_update: WidgetCallback::new(|text_input| PropertiesPanelMessage::ModifySolidFill { value: text_input.value.clone() }.into()),
+					})),
+				],
+			}],
+		},
+		Fill::LinearGradient(gradient) => LayoutRow::Section {
+			name: "Fill".into(),
+			layout: vec![
+				LayoutRow::Row {
+					name: "".into(),
+					widgets: vec![
+						WidgetHolder::new(Widget::TextLabel(TextLabel {
+							value: "0%".into(),
```

Comment:
I've been thinking about the best way to label this and I came up with `Gradient: 0%` and `Gradient: 100%`.